amespahdbpythonsuite/spectrum.py

```diff
@@ -362,3 +363,94 @@ def resample(self, grid: np.ndarray) -> None:
             self.data[uid] = s.flux.value
 
         self.grid = grid
+
+    def mcfit(self, obs, nsamples, **keywords) -> Mcfitted:
+        """
+        Monte Carlo sampling and fitting to the input spectrum.
+
+        Properties
+        ----------
+        obs : observations object
+
+        nsamples : Number of samples.
+            int
+
+        """
+        from amespahdbpythonsuite.mcfitted import Mcfitted
+
+        # Initialize lists and dictionaries.
+        fits = []
+        classes = []
```

Comment:
Conform the IDL-interface, which offers some 'Lazy' optimizations, move computing the breakdown, etc. into the MCFitted-class, which can be done on request, i.e., if only the breakdown is wanted, there is no need to compute the classes.